src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Diagnostics;
+
+namespace Datadog.Trace.Util
+{
+    /// <summary>
+    /// Dedicated helper class for consistently referencing Process and AppDomain information.
+    /// </summary>
+    internal static class DomainMetadata
+    {
+        private const string UnknownName = "unknown";
+        private static Process _currentProcess = null;
+        private static bool _processDataPoisoned = false;
```

Comment:
Does process and domain data actually ever get "poisoned"?

Reply:
Theoretically, accessing the Process or AppDomain can throw exceptions.

Reply:
Seemed good to be defensive.